servers/cromwell/jobs/util.py

```diff
@@ -23,11 +31,10 @@ def _deserialize(data, klass):
         return deserialize_date(data)
     elif klass == datetime:
         return deserialize_datetime(data)
-    elif type(klass) == GenericMeta:
-        if klass.__extra__ == list:
-            return _deserialize_list(data, klass.__args__[0])
-        if klass.__extra__ == dict:
-            return _deserialize_dict(data, klass.__args__[1])
+    elif str(typing.List) in str(klass):
+        return _deserialize_list(data, klass.__args__[0])
+    elif str(typing.Dict) in str(klass):
+        return _deserialize_dict(data, klass.__args__[1])
     else:
```

Comment:
This seems like a totally fine solution to me. I would guess there's a better way to do it, but <shrug>. I'm curious, though, is it the `GenericMeta` part that no longer worked? Or the `__extra__`?

Reply:
So at some level both, the __extra__ field I think is unique to either GenericMeta, or the List and Dict classes so we can't check for __extra__ and avoid errors for other classes it seemed. The new list and dict classes do not have the common base class GenericMeta anymore, so this check always failed. I tried some other strategies to just check if the class was a List or a Dict, but the template type meant that it would not match, I'd have to check for the possible sub types, which defeats the purpose of it being generic. 

This is a good reminder though I think I can remove the GenericMeta above, since I am no longer using it.